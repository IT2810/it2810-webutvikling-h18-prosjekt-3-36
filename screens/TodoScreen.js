import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  Alert,
} from 'react-native';
import TodoCalendar from "../components/TodoCalendar";
import TodoList from "../components/TodoList";
import TodoFormModal from "../components/TodoFormModal";

export default class TodoScreen extends React.Component {
  static navigationOptions = {
    title: 'Todo View',
    // header: null,
  };

  constructor(props) {
    super(props);
    this.state = {
      showCalendar: false,
      nextId: 3,
      todos: [{ id: 1, text: "kjøp melk", date: "2018-10-04", category: "work", completed: false }, { id: 2, text: "gjør webdev og balbajsklfjkalsdfjkla sdjklfja skldfj klasjdfkl jaklsdfjkla jsdklfj askldjfkl jasdklfj klasjdfk jaskldfjklasj klfjaskld fjklasdfjkl ", date: "2018-10-05", category: "work", completed: true }
      ]
    };
  }
  addNewTodo = (text, date) => {
    console.log(text + date);
    this.setState(prevState => ({
      todos: [...prevState.todos, {"id": prevState.nextId, "text": text, "date": date, "category": "work", "completed": false}],
      nextId: prevState.nextId + 1
    }));
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.calOrList}>
          {!this.state.showCalendar && <TodoList todos={this.state.todos} />}
          {this.state.showCalendar && <TodoCalendar/>}
        </View>
        <TodoFormModal addNewTodo={this.addNewTodo} style={styles.btnAndModal} />
      </View>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  },
  calOrList: {
    position: 'absolute',
    width: '100%'
  },
  btnAndModal: {
    position: 'absolute'
  }
});