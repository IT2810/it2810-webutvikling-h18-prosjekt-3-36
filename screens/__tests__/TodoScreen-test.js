import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoScreen from "../TodoScreen";
import { Provider } from 'react-redux'
import appState from '../../reducers/Reducer'
import { createStore } from 'redux'

const store = createStore(appState)


it("TodoScreen Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Provider store={store}><TodoScreen navigation={{}} /></Provider>)
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});
