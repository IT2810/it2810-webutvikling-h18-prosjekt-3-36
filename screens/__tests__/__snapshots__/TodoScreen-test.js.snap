// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoScreen Matches snapshot. 1`] = `
<TodoScreen
  navigation={Object {}}
/>
`;
